File: backend/src/main/java/com/waypoint/model/Stage.java
package com.waypoint.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Pipeline stage for an applicant. The JSON label (e.g. "Call scheduled") matches the
 * strings the frontend uses in {@code data.ts}, so the API and UI share one vocabulary.
 */
public enum Stage {
    APPLIED("Applied"),
    REVIEWED("Reviewed"),
    CALL_SCHEDULED("Call scheduled"),
    SELECTED("Selected");

    private final String label;

    Stage(String label) {
        this.label = label;
    }

    @JsonValue
    public String label() {
        return label;
    }

    @JsonCreator
    public static Stage fromLabel(String value) {
        for (Stage stage : values()) {
            if (stage.label.equalsIgnoreCase(value)) {
                return stage;
            }
        }
        throw new IllegalArgumentException("Unknown stage: " + value);
    }
}

File: frontend/src/lib/stage.ts
import type { Stage } from "./data";

interface StageStyle {
  text: string;
  dot: string;
  bar: string;
}

export const stageStyle: Record<Stage, StageStyle> = {
  Applied: { text: "text-stone", dot: "bg-stone", bar: "bg-stone" },
  Reviewed: { text: "text-gold", dot: "bg-gold", bar: "bg-gold" },
  "Call scheduled": { text: "text-gold", dot: "bg-gold", bar: "bg-gold" },
  Selected: { text: "text-forest-600", dot: "bg-forest-600", bar: "bg-forest-600" },
};
